libfwupd/fwupd-hsi-attr.c

```diff
@@ -0,0 +1,730 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+//#include <glib-object.h>
+#include <gio/gio.h>
+//#include <string.h>
+
+#include "fwupd-common-private.h"
+#include "fwupd-enums-private.h"
+//#include "fwupd-error.h"
+#include "fwupd-hsi-attr-private.h"
+
+/**
+ * SECTION:fwupd-hsi-attr
+ *
+ * An object that represents an Host Security ID attribute.
+ */
+
+static void fwupd_hsi_attr_finalize	 (GObject *object);
+
+typedef struct {
+	gchar				*appstream_id;
+	GPtrArray			*obsoletes;
+	gchar				*name;
+	gchar				*summary;
+	gchar				*uri;
+	guint32				 number;
+	FwupdHsiAttrFlags		 flags;
+} FwupdHsiAttrPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (FwupdHsiAttr, fwupd_hsi_attr, G_TYPE_OBJECT)
```

Comment:
I'm not 100% about this object name if you have any better ideas...

Reply:
FwupdSecurityAttr?

Reply:
I've used that, thanks.